src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,564 @@
+
+            return true;
+        }
+
+        private void RaiseUnknownHashingAlgorithm(HashingAlgorithm algorithm)
+        {
+            if (Log.IsWarnEnabled)
+                Log.WarnFormat("Unknown hashing algorithm number 0x{0:x2}.", algorithm.Code);
+
+            var handler = UnknownHashingAlgorithm;
+            if (handler != null)
+            {
+                var args = new UnknownHashingAlgorithmEventArgs
+                {
+                    Algorithm = algorithm
+                };
+                handler(this, args);
+            }
+        }
+
+        /// <summary>
+        ///   Generate the multihash for the specified byte array. 
+        /// </summary>
+        /// <param name="data">
+        ///   The byte array containing the data to hash.
+        /// </param>
+        /// <param name="algorithmName">
+        ///   The name of the hashing algorithm to use; defaults to <see cref="DefaultAlgorithmName"/>.
+        /// </param>
+        /// <returns>
+        ///   A <see cref="MultiHash"/> for the <paramref name="data"/>.
+        /// </returns>
+        public static MultiHash ComputeHash(byte[] data, string algorithmName = DefaultAlgorithmName)
+        {
+            using (var alg = GetHashAlgorithm(algorithmName))
+            {
+                return new MultiHash(algorithmName, alg.ComputeHash(data));
+            }
+        }
+
+        /// <summary>
+        ///   Generate the multihash for the specified <see cref="Stream"/>. 
+        /// </summary>
+        /// <param name="data">
+        ///   The <see cref="Stream"/> containing the data to hash.
+        /// </param>
+        /// <param name="algorithmName">
+        ///   The name of the hashing algorithm to use; defaults to <see cref="DefaultAlgorithmName"/>.
+        /// </param>
+        /// <returns>
+        ///   A <see cref="MultiHash"/> for the <paramref name="data"/>.
+        /// </returns>
+        public static MultiHash ComputeHash(Stream data, string algorithmName = DefaultAlgorithmName)
+        {
+            using (var alg = GetHashAlgorithm(algorithmName))
+            {
+                return new MultiHash(algorithmName, alg.ComputeHash(data));
+            }
+        }
+
+        /// <summary>
+        ///   Conversion of a <see cref="MultiHash"/> to and from JSON.
+        /// </summary>
+        /// <remarks>
+        ///   The JSON is just a single string value.
+        /// </remarks>
+        private class Json : JsonConverter
+        {
+            public override bool CanConvert(Type objectType) { return true; }
+
+            public override bool CanRead => true;
+            public override bool CanWrite => true;
+
+            public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+            {
+                var mh = value as MultiHash;
+                writer.WriteValue(mh?.ToString());
+            }
+
+            public override object ReadJson(JsonReader reader,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'ReadJson'.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)